r2/r2/lib/base.py

```diff
@@ -43,12 +44,17 @@
 logging.getLogger('scgi-wsgi').setLevel(logging.CRITICAL)
 
 
-def is_local_address(ip):
-    if hasattr(g, 'trusted_proxy_ips'):
-        trusted_ips = [x.strip() for x in g.trusted_proxy_ips.split(',')]
-    else:
-        trusted_ips = []
-    return ip.startswith('10.') or ip == "127.0.0.1" or ip in trusted_ips
+def is_trusted_proxy(ip):
+    ip = ipaddress.IPv4Address(ip)
+    trusted_ips = getattr(g, 'trusted_proxy_ips', '').strip()
+    if not trusted_ips:
+        return False
+    trusted_ips = [x.strip() for x in trusted_ips.split(',')]
+    trusted_ips = [ipaddress.IPv4Network(x.strip(), False) for x in trusted_ips]
```

Comment:
This can be kinda expensive to do, it'd be better to pre-build this list of objects in `app_globals` so it's not re-running on every request.